test/integration/scheduler/preemption_test.go

```diff
@@ -434,6 +437,97 @@ func TestPreemption(t *testing.T) {
 	}
 }
 
+// TestNonPreemption tests NonPreempt option of PriorityClass of scheduler works as expected.
+func TestNonPreemption(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.NonPreemptingPriority, true)()
+
+	var preemptNever = v1.PreemptNever
+	// Initialize scheduler.
+	testCtx := initTest(t, "non-preemption")
+	defer testutils.CleanupTest(t, testCtx)
+	cs := testCtx.ClientSet
+	period := int64(0)
+
+	victim := initPausePod(testCtx.ClientSet, &pausePodConfig{
```

Comment:
we usually embed the pods into the test struct to make sure they're not shared across tests - incase some test modifies them.